### ISSUE TITLE
dfdaemon HTTPS protocol is not valid

### ISSUE THREAD

**Issue by readerx** [dragonflyoss/Dragonfly#1231]
### Describetion:

dfclient config：

```
registry_mirror:
  remote: https://192.168.1.254:5000
  insecure: false
  certs:
  - /certs/ca-registry.crt

port: 8443
keypem: /certs/dragonfly.key
certpem: /certs/dragonfly.cer
```

docker command:

```
docker run -d --name dfclient -p 443:8443 \
    -v `pwd`/certs:/certs \
    -v `pwd`/dfclient.yaml:/dfclient.yaml \
    dragonflyoss/dfclient:1.0.0 --verbose --config=/dfclient.yaml
```

docker pull error:

```
Error response from daemon: Get https://192.168.1.111/v2/: http: server gave HTTP response to HTTPS client
```

error code in dfdaemon/server.go:

```
// Start runs dfdaemon's http server.
func (s *Server) Start() error {
	_ = proxy.WithDirectHandler(handler.New())(s.proxy)
	s.server.Handler = s.proxy
	if s.server.TLSConfig != nil {
		logrus.Infof("start dfdaemon https server on %s", s.server.Addr)
	} else {
		logrus.Infof("start dfdaemon http server on %s", s.server.Addr)
	}

	return s.server.ListenAndServe()
}
```

###  Environment:

- dragonfly version: 1.0.0

**Comment by readerx** [comment]
@lowzj

**Comment by readerx** [comment]
It was solved by #1232
